omegaconf/basecontainer.py

```diff
@@ -103,7 +103,19 @@ def __setstate__(self, d: Dict[str, Any]) -> None:
 
         if isinstance(self, DictConfig):
             key_type = d["_metadata"].key_type
+
+            # backward compatibility to load OmegaConf 2.0 configs
+            if key_type is None:
+                key_type = Any
+                d["_metadata"].key_type = key_type
+
         element_type = d["_metadata"].element_type
+
+        # backward compatibility to load OmegaConf 2.0 configs
+        if element_type is None:
+            element_type = Any
+            d["_metadata"].element_type = element_type
+
```

Comment:
Would it make sense to also raise a deprecation warning asking people re-serialize their configuration? (the warning could link to an issue showing how to do it)
Unless the intention is to keep such backward compatibility forever?

Reply:
We can consider deprecating in the future. I think over time this will become irrelevant as most of the checkpoints will either be lost or re-created anyway.